File: functions/golang/logic.go
package functions

func Compute(items []int) int {
	sum := 0
	for _, num := range items {
		sum += num
	}
	return sum
}

func Multiply(items []int) int {
	result := 1
	for _, num := range items {
		result *= num
	}
	return result
}

func Average(items []int) float64 {
	if len(items) == 0 {
		return 0
	}
	sum := Compute(items)
	return float64(sum) / float64(len(items))
}

File: functions/python/logic.py
def compute(numbers: list[int]):
    """Computes the sum of numbers."""
    return sum(numbers)


def multiply(numbers: list[int]):
    """Multiplies all numbers together."""
    result = 1
    for num in numbers:
        result *= num
    return result


def average(numbers: list[int]):
    """Calculates the average of numbers."""
    return sum(numbers) / len(numbers) if numbers else 0
